File: Baekjoon/B2096.java
import java.io.*;
import java.util.StringTokenizer;

public class B2096 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int rowCnt = Integer.parseInt(st.nextToken());
        int[] min = new int[3];
        int[] max = new int[3];
        int[] layer = new int[3];
        for (int i=0; i<rowCnt; i++){
            st = new StringTokenizer(br.readLine());
            layer[0] = Integer.parseInt(st.nextToken());
            layer[1] = Integer.parseInt(st.nextToken());
            layer[2] = Integer.parseInt(st.nextToken());
            if (i == 0) {
                max[0] = layer[0];
                max[1] = layer[1];
                max[2] = layer[2];
                min[0] = layer[0];
                min[1] = layer[1];
                min[2] = layer[2];
            }
            else{
                minSum(min, layer);
                maxSum(max, layer);
            }
        }
        System.out.printf("%d %d",Math.max(max[0], Math.max(max[1], max[2])), Math.min(min[0], Math.min(min[1], min[2])));
    }

    public static void minSum(int[] min,int[] layer){
        int min0 = layer[0] + Math.min(min[0], min[1]);
        int min1 = layer[1] + Math.min(min[0], Math.min(min[1], min[2]));
        int min2 = layer[2] + Math.min(min[1], min[2]);
        min[0] = min0;
        min[1] = min1;
        min[2] = min2;
    }

    public static void maxSum(int[] max,int[] layer){
        int max0 = layer[0] + Math.max(max[0], max[1]);
        int max1 = layer[1] + Math.max(max[0], Math.max(max[1], max[2]));
        int max2 = layer[2] + Math.max(max[1], max[2]);
        max[0] = max0;
        max[1] = max1;
        max[2] = max2;
    }

}

File: Baekjoon/JS/B2096.js
const readline = require('readline');

const rl = readline.createInterface({
	input: process.stdin,
	output: process.stdout
});

let cnt;
let rowNum=0;
let max;
let min;

rl.on('line', function (line){
	if (rowNum === 0){
		cnt = Number(line);
		rowNum++;
	}
	else if (rowNum === 1){
		min = line.split(' ').map(el => Number(el));
		max = line.split(' ').map(el => Number(el));
		rowNum++;
	}
	else{
		let layer = line.split(' ').map(el => Number(el));
		minSum(min, layer);
		maxSum(max, layer);
	}
})
.on('close',function(){
	console.log(`${Math.max(...max)} ${Math.min(...min)}`);
	process.exit();
});

function minSum(min, layer){
	min[0] = Math.min(min[0] + layer[0], min[0] + layer[1]);
	min[1] = Math.min(min[1] + layer[0], min[1] + layer[1], min[1] + layer[2]);
	min[2] = Math.min(min[2] + layer[1], min[2] + layer[2]);
}

function maxSum(max, layer){
	max[0] = Math.max(max[0] + layer[0], max[0] + layer[1]);
	max[1] = Math.max(max[1] + layer[0], max[1] + layer[1], max[1] + layer[2]);
	max[2] = Math.max(max[2] + layer[1], max[2] + layer[2]);
}